cpp/include/cugraph/prims/gather_edges.cuh

```diff
@@ -0,0 +1,710 @@
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param graph_view Non-owning graph object.
+ * @param active_sources Device vector containing all the vertex id that are processed by
+ * gpus in the column communicator
+ * @param global_out_degrees Global out degrees for every source represented by current gpu
+ * @return Global out degrees of all sources in active_sources
+ */
+template <typename GraphViewType>
+rmm::device_uvector<typename GraphViewType::edge_type> get_active_source_global_degrees(
+  raft::handle_t const& handle,
+  GraphViewType const& graph_view,
+  rmm::device_uvector<typename GraphViewType::vertex_type>& active_sources,
+  const rmm::device_uvector<typename GraphViewType::edge_type>& global_out_degrees)
+{
+  using vertex_t    = typename GraphViewType::vertex_type;
+  using edge_t      = typename GraphViewType::edge_type;
+  using partition_t = matrix_partition_device_view_t<typename GraphViewType::vertex_type,
+                                                     typename GraphViewType::edge_type,
+                                                     typename GraphViewType::weight_type,
+                                                     GraphViewType::is_multi_gpu>;
+  rmm::device_uvector<edge_t> active_source_degrees(active_sources.size(), handle.get_stream());
+
+  std::vector<vertex_t> id_begin;
+  std::vector<vertex_t> id_end;
+  std::vector<vertex_t> count_offsets;
+  id_begin.reserve(graph_view.get_number_of_local_adj_matrix_partitions());
+  id_end.reserve(graph_view.get_number_of_local_adj_matrix_partitions());
+  count_offsets.reserve(graph_view.get_number_of_local_adj_matrix_partitions());
+  vertex_t counter{0};
+  for (size_t i = 0; i < graph_view.get_number_of_local_adj_matrix_partitions(); ++i) {
+    auto matrix_partition = partition_t(graph_view.get_matrix_partition_view(i));
+    // Starting vertex ids of each partition
+    id_begin.push_back(graph_view.get_local_adj_matrix_partition_row_first(i));
+    id_end.push_back(graph_view.get_local_adj_matrix_partition_row_last(i));
+    count_offsets.push_back(counter);
+    counter += matrix_partition.get_major_size();
+  }
+  rmm::device_uvector<vertex_t> vertex_id_begin(id_begin.size(), handle.get_stream());
+  rmm::device_uvector<vertex_t> vertex_id_end(id_end.size(), handle.get_stream());
+  rmm::device_uvector<vertex_t> vertex_count_offsets(count_offsets.size(), handle.get_stream());
+  raft::update_device(
+    vertex_id_begin.data(), id_begin.data(), id_begin.size(), handle.get_stream());
+  raft::update_device(vertex_id_end.data(), id_end.data(), id_end.size(), handle.get_stream());
+  raft::update_device(
+    vertex_count_offsets.data(), count_offsets.data(), count_offsets.size(), handle.get_stream());
+
+  auto& comm           = handle.get_comms();
+  auto const comm_rank = comm.get_rank();
```

Comment:
So, is this code intended to use only for the multi-GPU cases? If not, this should be guarded inside `if constepxr (GraphViewType::is_multi_gpu)`. If to be used only for the multi-GPU use cases, there should be either something like `std::enable_if<GraphViewType::is_multi_gpu>` or `static_assert(GrpahViewType::is_multi_gpu);`

Reply:
What you need is just comm_rank, you can do `auto const comm_rank = GraphViewType::is_multi_gpu ? handle.get_comms().get_rank() : int{0};`.

`handle.get_comms().get_rank()` will seg-fault in single-GPU.